src/Pages/Frontpage/SignUpForm.js

```diff
@@ -1,27 +1,81 @@
-import React from 'react'
-import { useForm } from 'react-hook-form'
-import { yupResolver } from '@hookform/resolvers'
-import * as yup from "yup"
+import React, { useState, useContext } from "react";
+import { UserContext } from "../../Main/UserContext";
+import { useForm } from "react-hook-form";
+import { yupResolver } from "@hookform/resolvers";
+import * as yup from "yup";
+
 
 export default () => {
+  const [name, setName] = useState("");
+  const [email, setEmail] = useState("");
+  const [password, setPassword] = useState("");
+  const fb = useContext(UserContext).firebase;
+  const [loggedIn, setLoggedIn] = useContext(UserContext).loggedIn;
 
-    const schema = yup.object().shape({
-        name: yup.string().required(),
-        email: yup.string().required(),
-        password: yup.string().required()
-    })
+  const schema = yup.object().shape({
+    name: yup.string().required(),
+    email: yup.string().required(),
+    password: yup.string().required(),
+  });
 
-    const { register, handleSubmit } = useForm({
-        resolver: yupResolver(schema),
-    })
+  const { register, handleSubmit } = useForm({
+    resolver: yupResolver(schema),
+  });
 
+  return (
+      <form
+        className="sign-up-form"
+        onSubmit={(e) => e.preventDefault() && false}
+      >
+        <div className="SignUpH">
+        <h1>Sign up</h1>
+      </div>
+      <div className="Inputs">
+        <div className="Name">
+          <p>Name</p>
+          <input
+            id="name"
+            name="name"
+            type="text"
+            value={name}
+            onChange={(e) => setName(e.target.value)}
+            ref={register}
+        />
+        </div>
+         <div className="Email">
+          <p>Email</p>
+          <input
+            id="email"
+            name="email"
+            type="email"
+            value={email}
+            onChange={(e) => setEmail(e.target.value)}
+            ref={register}
+          />
+        </div>
+        <div className="Password">
+        <p>Password</p>
+          <input
+            id="password"
+            name="password"
+            type="text"
```

Comment:
type should be password so that text is marked with dots, not visible to others as all password fields are implemented that way.

Reply:
Should probably have commented on this in the code.
This was intentionally type text so that we could see what we wrote during testing.
Will be changed to password for final version.